pkg/forwarder/transactions_serializer.go

```diff
@@ -47,9 +71,9 @@ func (s *TransactionsSerializer) Add(transaction *HTTPTransaction) error {
 
 	endpoint := transaction.Endpoint
 	transactionProto := HttpTransactionProto{
-		Domain:     transaction.Domain,
-		Endpoint:   &EndpointProto{Route: endpoint.route, Name: endpoint.name},
-		Headers:    toHeaderProto(transaction.Headers),
+		Domain:     "", // Domain is always the same
```

Comment:
Even though it's the same domain, why not include it? To save space? Do we compress when we write to disk? If we compress the payloads, then including the `Domain` should have negligible impact.

Reply:
It is mostly for security reasons. If a user can update the domain from the serialized transactions, they can replace the domain by `127.0.0.1:1234` and then the Agent will send the HTTP transactions to `127.0.0.1:1234` on an endpoint containing the API KEY.

Reply:
I think this should be explained in a comment inline and tested (I didn't see a test for this but may have missed something), otherwise this crucial context will be lost in the future